install/kubernetes/cilium/templates/cilium-operator/deployment.yaml

```diff
@@ -18,10 +18,22 @@ spec:
     matchLabels:
       io.cilium/app: operator
       name: cilium-operator
+  # ensure operator update on single node k8s clusters, by using rolling update with maxUnavailable=100% in case
+  # of one replica and no user configured Recreate strategy.
+  # otherwise an update might get stuck due to the default maxUnavailable=50% in combination with the
+  # podAntiAffinity which prevents deployments of multiple operator replicas on the same node.
+  {{- if and (eq (.Values.operator.replicas | toString) "1") (eq .Values.operator.updateStrategy.type "RollingUpdate") }}
```

Comment:
do we need to convert toString for comparison ?

Reply:
yes, otherwise comparison might fail when the user is providing the replicas as integer (which is possible)